pkg/kubernetes-mcp-server/cmd/root.go

```diff
@@ -1,140 +1,181 @@
-	rootCmd.Flags().IntP("sse-port", "", 0, "Start a SSE server on the specified port")
-	rootCmd.Flags().IntP("http-port", "", 0, "Start a streamable HTTP server on the specified port")
-	rootCmd.Flags().StringP("sse-base-url", "", "", "SSE public base URL to use when sending the endpoint message (e.g. https://example.com)")
-	rootCmd.Flags().StringP("kubeconfig", "", "", "Path to the kubeconfig file to use for authentication")
-	rootCmd.Flags().String("profile", "full", "MCP profile to use (one of: "+strings.Join(mcp.ProfileNames, ", ")+")")
-	rootCmd.Flags().String("list-output", "table", "Output format for resource list operations (one of: "+strings.Join(output.Names, ", ")+")")
-	rootCmd.Flags().Bool("read-only", false, "If true, only tools annotated with readOnlyHint=true are exposed")
-	rootCmd.Flags().Bool("disable-destructive", false, "If true, tools annotated with destructiveHint=true are disabled")
-	_ = viper.BindPFlags(rootCmd.Flags())
+func (m *MCPServerOptions) Validate() error {
+	return nil
 }
 
-func init() {
-	flagInit()
+func (m *MCPServerOptions) Run() error {
+	profile := mcp.ProfileFromString(m.Profile)
+	if profile == nil {
+		return fmt.Errorf("Invalid profile name: %s, valid names are: %s\n", m.Profile, strings.Join(mcp.ProfileNames, ", "))
+	}
+	listOutput := output.FromString(m.ListOutput)
+	if listOutput == nil {
+		return fmt.Errorf("Invalid output name: %s, valid names are: %s\n", m.ListOutput, strings.Join(output.Names, ", "))
+	}
+	klog.V(1).Info("Starting kubernetes-mcp-server")
+	klog.V(1).Infof(" - Profile: %s", profile.GetName())
+	klog.V(1).Infof(" - ListOutput: %s", listOutput.GetName())
+	klog.V(1).Infof(" - Read-only mode: %t", m.ReadOnly)
+	klog.V(1).Infof(" - Disable destructive tools: %t", m.DisableDestructive)
+
+	if m.Version {
+		fmt.Fprintf(m.Out, "%s\n", version.Version)
+		return nil
+	}
+	mcpServer, err := mcp.NewSever(mcp.Configuration{
+		Profile:            profile,
+		ListOutput:         listOutput,
+		ReadOnly:           m.ReadOnly,
+		DisableDestructive: m.DisableDestructive,
+		Kubeconfig:         m.Kubeconfig,
+	})
+	if err != nil {
+		return fmt.Errorf("Failed to initialize MCP server: %w\n", err)
+	}
+	defer mcpServer.Close()
+
+	ctx := context.Background()
+
+	if m.SSEPort > 0 {
+		sseServer := mcpServer.ServeSse(m.SSEBaseUrl)
+		defer func() { _ = sseServer.Shutdown(ctx) }()
+		klog.V(0).Infof("SSE server starting on port %d and path /sse", m.SSEPort)
+		if err := sseServer.Start(fmt.Sprintf(":%d", m.SSEPort)); err != nil {
+			return fmt.Errorf("failed to start SSE server: %w\n", err)
+		}
+	}
+
+	if m.HttpPort > 0 {
+		httpServer := mcpServer.ServeHTTP()
+		klog.V(0).Infof("Streaming HTTP server starting on port %d and path /mcp", m.HttpPort)
+		if err := httpServer.Start(fmt.Sprintf(":%d", m.HttpPort)); err != nil {
+			return fmt.Errorf("failed to start streaming HTTP server: %w\n", err)
+		}
+	}
+
+	if err := mcpServer.ServeStdio(); err != nil && !errors.Is(err, context.Canceled) {
```

Comment:
Since we don't pass context in anywhere, it is not possible to get `context.Cancelled`. We need to wire this.